lib/ansible/modules/user.py

```diff
@@ -1372,14 +1372,25 @@ def create_homedir(self, path):
             if os.path.exists(self.LOGIN_DEFS):
                 with open(self.LOGIN_DEFS, 'r') as f:
                     for line in f:
+                        # HOME_MODE has higher precedence as UMASK
+                        m = re.match(r'^HOME_MODE\s+(\d+)$', line)
+                        if m:
+                            mode = int(m.group(1), 8)
+                            break  # higher precedence
                         m = re.match(r'^UMASK\s+(\d+)$', line)
                         if m:
                             umask = int(m.group(1), 8)
                             mode = 0o777 & ~umask
```

Comment:
```suggestion
                            mode = 0o777 & ~umask
                            break
```

Reply:
I deliberately left the `break` out. If `UMASK` happens to match first, and `HOME_MODE` is set later in the file, the latter will be ignored. It however should have higher precedence to match the behaviour of `adduser`.